### PR TITLE
wait for commit to be in git repository

### PR DESCRIPTION
**Pull request by grosser**
trying to fix issue found in https://github.com/zendesk/samson/pull/3747 after https://github.com/zendesk/samson/pull/3748 did not fully fix the issue

/cc @zendesk/compute @patrobinson

### CODE REVIEW

File: test/models/job_execution_test.rb [hunk 1]
```diff
@@ -251,7 +251,7 @@ def last_line_of_output
   end
 
   it 'errors if job commit resultion fails, but checkout works' do
```
Comment (adammw) [inline test/models/job_execution_test.rb]:
```suggestion
  it 'errors if job commit resolution fails, but checkout works' do
```

File: test/models/job_execution_test.rb [hunk 2]
```diff
@@ -251,7 +251,7 @@ def last_line_of_output
...
-    GitRepository.any_instance.expects(:commit_from_ref).returns nil
+    GitRepository.any_instance.expects(:commit_from_ref).times(4).returns nil
```
Comment (adammw) [inline test/models/job_execution_test.rb]: separate test for fails 3 times then works?

**Comment by patrobinson** [comment]
@grosser Unfortunately this doesn't fix the issue. I deployed this commit to our Production environment and it's still happening.

Screen Shot 2020-03-30 at 12 29 56 pm

**Comment by grosser** [comment]
ahhh ... it needs to update the mirror :D

**Comment by grosser** [comment]
hope this does it ... 1 more test plz 🤞

**Comment by patrobinson** [comment]
@grosser Confirmed this is working now, thanks for your help

**Comment by grosser** [comment]
awesome! :D

**Comment by grosser** [comment]
I guess the root issue was that when making a new tag the api will know about it, but it takes a few seconds to be available in git ...

**Pull request merged**